dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,567 @@
+
+        Args:
+            table_name: The name of the table to delete.
+        """
+        self.db_client.drop_table(table_name)
+
+    def delete_all_tables(self) -> None:
+        """Delete all LanceDB tables from the LanceDB instance and all data associated with it."""
+        self.db_client.drop_database()
+
+    def query_table(
+        self,
+        table_name: str,
+        query: Union[
+            List[Any], ndarray[Any, Any], Array, ChunkedArray, str, Tuple[Any], None
+        ] = None,
+    ) -> LanceQueryBuilder:
+        """Query a LanceDB table.
+
+        Args:
+            table_name: The name of the table to query.
+            query: The targeted vector to search for.
+
+        Returns:
+            A LanceDB query builder.
+        """
+        return self.db_client.open_table(table_name).search(query=query)
+
+    def add_to_table(
+        self,
+        table_name: str,
+        data: DATA,
+        mode: str = "append",
+        on_bad_vectors: str = "error",
+        fill_value: float = 0.0,
+    ) -> None:
+        """Add more data to the LanceDB Table.
+
+        Args:
+            table_name (str): The name of the table to add data to.
+            data (DATA): The data to insert into the table.
+                         Acceptable types are:
+                            - dict or list-of-dict
+                            - pandas.DataFrame
+                            - pyarrow.Table or pyarrow.RecordBatch
+            mode (str): The mode to use when writing the data.
+            Valid values are
+                "append" and "overwrite".
+            on_bad_vectors (str): What to do if any of the vectors are different
+                size or contain NaNs.
+                One of "error", "drop", "fill".
+                Defaults to
+                "error".
+            fill_value (float): The value to use when filling vectors.
+            Only used if
+                on_bad_vectors="fill".
+                Defaults to 0.0.
+
+        Returns:
+            None
+        """
+        self.db_client.open_table(table_name).add(data, mode, on_bad_vectors, fill_value)
+
+    @lancedb_error
+    def drop_storage(self) -> None:
+        """Drop the dataset from the LanceDB instance.
+
+        Deletes all tables in the dataset and all data, as well as sentinel table associated with them.
+
+        If the dataset name was not provided, it deletes all the tables in the current schema.
+        """
+        if self.dataset_name:
+            prefix = f"{self.dataset_name}{self.config.dataset_separator}"
+            table_names = [
+                table_name
+                for table_name in self.db_client.table_names()
+                if table_name.startswith(prefix)
+            ]
+        else:
+            table_names = self.db_client.table_names()
+
+        for table_name in table_names:
+            self.db_client.drop_table(table_name)
+
+        self._delete_sentinel_table()
```

Comment:
Please make sure that when creating and deleting the sentinel table, this table is prefixed with the datasetname. Every "virtual" dataset in this database needs its own sentinel table, otherwise stuff will stop working!

Reply:
Actually I think you are doing this, but please do it the same way we do it in the other destinations that do not have datasets as a concept. The sentinel table should always be {dataset_name}{separator}{sentinel_table_name}

Reply:
Changed it to 

```python
    @property
    def sentinel_table(self) -> str:
        return self.make_qualified_table_name("dltSentinelTable")
```

Please advise if this is good.

Reply:
yes, thanks :)